### PR TITLE
Fix failed onednn tests

### PR DESCRIPTION
**Pull request by byungilm**
### Details:
 - Fix failed unit-tests on dGPU
 - Modified fully_connected_random_test_i8_3d not to have ambiguous input
 - oneDNN does NOT support i64 type.
 -  bugfix in prepare_primitive_fusing about exception of activation function
 - Add exception logic for dynamic to select ocl type in is_node_for_onednn 
 
### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 1]
```diff
@@ -700,14 +700,17 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
...
                     return;
-                #ifdef ENABLE_ONEDNN_FOR_GPU
-                // Activation should not fused if it isn't supported in onednn
-                try {
-                    onednn::convert_activation_func(activation_node.get_primitive()->activation_function);
-                } catch (...) {
-                    return;
+
+                // Activation should not be fused if oneDNN does NOT support it
+                if (_lo.are_primitives_implemented_for_onednn(input) && p.get_engine().get_device_info().supports_immad)  {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:705]: * `are_primitives` -> `is_primitive`
* no need to check `supports_immad` because we already checked `use_onednn_impls`

File: src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp [hunk 2]
```diff
@@ -223,6 +223,42 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, activation_eltwise_activation_quantize_u8,
...
+    activation_test_params{ CASE_ACTIVATION_F32_1, 3, 5, "activation_opt" },
+    activation_test_params{ CASE_ACTIVATION_3D_F32_0, 3, 5, "activation_opt" },
+    activation_test_params{ CASE_ACTIVATION_3D_F32_1, 3, 5, "activation_opt" },
+
+    activation_test_params{ CASE_ACTIVATION_F32_0, 3, 5, "activation_ref" },
+    activation_test_params{ CASE_ACTIVATION_F32_1, 3, 5, "activation_ref" },
+    activation_test_params{ CASE_ACTIVATION_3D_F32_0, 3, 5, "activation_ref" },
+    activation_test_params{ CASE_ACTIVATION_3D_F32_1, 3, 5, "activation_ref" },
+}));
+
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp:261]: is this change still necessary after updating in `prepare_primitive_fusing`?

File: src/plugins/intel_gpu/tests/fusions/lrn_fusion_test.cpp [hunk 3]
```diff
@@ -118,7 +118,7 @@ TEST_P(lrn_fp32_quantize_u8_eltwise_activation, basic) {
         quantize("quantize", input_info("lrn_norm"), input_info("in_lo"), input_info("in_hi"),
                  input_info("out_lo"), input_info("out_hi"), 256, data_types::u8),
         eltwise("eltwise", { input_info("quantize"), input_info("eltwise_data") }, eltwise_mode::prod),
-        activation("activation", input_info("eltwise"), activation_func::floor),
+        activation("activation", input_info("eltwise"), activation_func::relu),
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/lrn_fusion_test.cpp:121]: is this change still necessary after updating in prepare_primitive_fusing?

File: src/plugins/intel_gpu/tests/fusions/fully_connected_fusion_test.cpp [hunk 4]
```diff
@@ -278,9 +278,6 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, fc_int8_eltwise, ::testing::ValuesIn(std::
     fully_connected_test_params{ CASE_FC_U8S8_1, 2, 3 },
     fully_connected_test_params{ CASE_FC_U8S8_2, 2, 3 },
     fully_connected_test_params{ CASE_FC_U8S8_3, 2, 3 },
-    fully_connected_test_params{ CASE_FC_U8S8_3D_1, 2, 3 },
-    fully_connected_test_params{ CASE_FC_U8S8_3D_2, 2, 3 },
-    fully_connected_test_params{ CASE_FC_U8S8_3D_3, 2, 3 },
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/fully_connected_fusion_test.cpp:283]: why did you remove this?

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/tests/test_cases/fully_connected_gpu_test.cpp [hunk 5]
```diff
@@ -2169,9 +2170,18 @@ struct dynamic_fully_connected_gpu : ::testing::TestWithParam<fully_connected_dy
...
                                                                               weights_data_vec,
                                                                               bias_data_vec);
-            for (int b = 0; b < batch_size; b++) {
-                for (int ofm = 0; ofm < output_f; ofm++) {
-                    ASSERT_EQ(ref_result[b * output_f + ofm], output_ptr[b * output_f + ofm]);
+
+            if (engine.get_device_info().supports_immad) {
+                for (int b = 0; b < batch_size; b++) {
+                    for (int ofm = 0; ofm < output_f; ofm++) {
+                        EXPECT_NEAR(ref_result[b * output_f + ofm], output_ptr[b * output_f + ofm], 0.5);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_cases/fully_connected_gpu_test.cpp:2177]: 0.5 seems quite large. What is the output data range? What about using `get_default_tolerance` function to set different threshold based on data type?

Comment (byungilm) [reply src/plugins/intel_gpu/tests/test_cases/fully_connected_gpu_test.cpp:2177]: Ok I will apply this fix.
If data_types are s8, s8, fp32 for input0, weight and output, then accuracy type is s32.
I think this makes main difference. So for now, I will use default_tolerance with input_type.

Comment (byungilm) [reply src/plugins/intel_gpu/tests/test_cases/fully_connected_gpu_test.cpp]: Fixed

**Review by isanghao — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 6]
```diff
@@ -1259,6 +1263,16 @@ bool layout_optimizer::are_layouts_suitable_for_onednn(program_node& node) {
     return true;
 }
 
+bool layout_optimizer::is_primitive_implemented_for_onednn(program_node& node) {
+    if (node.is_type<fully_connected>() || node.is_type<gemm>() || node.is_type<pooling>() ||
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1267]: How about add this querying function in primitive_type_base.h : 
as `does_possible_implementation_exist(const cldnn::program_node& node, const kernel_impl_params& impl_param, impl_types impl_type) ?`

Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1267]: Sounds good. Shall we follow-up on the next PR @byungilm ?
